gammapy/modeling/models/cube.py

```diff
@@ -77,15 +79,16 @@ def __init__(
                 "Spectral model used with SkyModel requires a norm type parameter."
             )
 
-        is_free_norm = np.array(
-            [par.is_norm for par in spectral_model.parameters.free_parameters]
-        )
-
-        if np.sum(is_free_norm) > 1:
-            raise ValueError(
-                "Spectral model used with SkyModel can only have a single free norm type parameter."
+        if isinstance(self.spectral_model, CompoundSpectralModel):
+            is_free_norm = np.array(
+                [par.is_norm for par in spectral_model.parameters.free_parameters]
             )
 
+            if self.spectral_model.operator is operator.mul and np.sum(is_free_norm) > 1:
+                raise ValueError(
+                    "The multiplicative compound spectral model can only have a single free norm type parameter."
```

Comment:
Thanks @bkhelifi, I'm not sure this is the best place to raise this error as the multiplication may make sense if the norm correction applies to a linked model, in that case there is no 100% correlation with the amplitude.
Imagine a joint fit of two dataset with a source fitted by two models, one for each dataset,  with linked amplitude but the second dataset have in addition a norm correction to compensate for systematic bias in flux within a given range (eventually linked between multiple sources for the second instrument).

So I think we should remove the error raise here and force the secondary norm parameters of a compound model to be frozen when reoptimize=True in the FluxPointsEstimator to solve the original problem. 

In a separate PR, the error raise could be introduced at the fit level in a specific method that verify the validity of the full parameters list (so there we can check for links) and we could eventually add other conditions to this method in future (no free parameters, value outside min/max range, bad parameter scaling with value below precision...).

Reply:
Thanks @QRemy, indeed this is a use-case I haven't thought of either. So +1 for your proposed solution...

Reply:
My initial thought was indeed to make nothing within the SkyModel, but only in the FluxPointEstimator.
For a 3D analysis, it think that the user could/should understand when a model can not be fit because of degeneracy.

So, what is the exact proposal?
- remove simply the currents line 82->89
- make an other PR for the FluxPointEstimator to raise an error when there is a degeneracy? I think that this is up to the user to choose what to froze and not up tp Gammapy to chose what/guess is 'secondary' parameter..

Reply:
Actually only the selected norm appears in the spectral model parameters after we apply the ScaleModel so there is mo issue with compound model.